### PR TITLE
Pr/m4lcd

### PR DESCRIPTION
**Pull request by gkalsi**

### CODE REVIEW

File: target/qemu-m4/m4display.c [hunk 1]
```diff
@@ -0,0 +1,217 @@
...
+#define SCENE_ERROR  0x03
+
+#define M4DISPLAY_WIDTH  180
+#define M4DISPLAY_HEIGHT 180
+
+static uint8_t framebuffer[M4DISPLAY_HEIGHT][M4DISPLAY_WIDTH];
+
+const char programming_header[] = "  Lattice\0iCEcube2 2014.08.26723\0Part: iCE40LP1K-CM36\0Date: Jan 30 2015 15:11:";
+
+void chip_select(bool val)
```
Comment (travisg) [inline target/qemu-m4/m4display.c]: static

File: target/qemu-m4/m4display.c [hunk 2]
```diff
@@ -0,0 +1,217 @@
...
+#define SCENE_SPLASH 0x01
+#define SCENE_UPDATE 0x02
+#define SCENE_ERROR  0x03
+
+#define M4DISPLAY_WIDTH  180
+#define M4DISPLAY_HEIGHT 180
+
+static uint8_t framebuffer[M4DISPLAY_HEIGHT][M4DISPLAY_WIDTH];
+
+const char programming_header[] = "  Lattice\0iCEcube2 2014.08.26723\0Part: iCE40LP1K-CM36\0Date: Jan 30 2015 15:11:";
```
Comment (travisg) [inline target/qemu-m4/m4display.c]: static

File: target/qemu-m4/rules.mk [hunk 3]
```diff
@@ -16,7 +16,11 @@ GLOBAL_DEFINES += \
     PKTBUF_POOL_SIZE=16
 
 MODULE_SRCS += \
+	$(LOCAL_DIR)/m4display.c \
```
Comment (travisg) [inline target/qemu-m4/rules.mk]: alphabetically sort these

File: target/qemu-m4/m4display.c [hunk 4]
```diff
@@ -0,0 +1,217 @@
...
+
+    SPI_Init(SPI6, &init_struct);
+
+    SPI_Cmd(SPI6, ENABLE); // enable SPI6
+
+    chip_select(true);
+    reset(true);
+    chip_select(false);
+
+    uint8_t draw_splash_cmds[] = { 0x04, 0x01 };
```
Comment (travisg) [inline target/qemu-m4/m4display.c]: can make these const, which should make the compiler not make a separate copy, though it likely matters not at all

**Comment by travisg** [comment]
couple of nits but otherwise LGTM

**Comment by gkalsi** [comment]
merged

**Pull request closed without merging**